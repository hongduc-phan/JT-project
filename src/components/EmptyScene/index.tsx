export * from './EmptyScene';
export {default} from './EmptyScene';
