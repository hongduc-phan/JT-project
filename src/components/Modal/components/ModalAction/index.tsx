export {default} from './ModalAction';
export * from './ModalAction';
