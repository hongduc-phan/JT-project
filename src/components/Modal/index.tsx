export {default} from './Modal';
export {default as ModalAction} from './components/ModalAction';
export * from './Modal';
export * from './components/ModalAction';
