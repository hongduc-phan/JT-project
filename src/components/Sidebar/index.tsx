export {default} from './Sidebar';
export {default as SidebarButton} from './SidebarButton';
export {default as SidebarMenu} from './SidebarMenu';
export {default as SidebarTitle} from './SidebarTitle';
export {default as SidebarMenuItem} from './SidebarMenuItem';
export {default as SidebarLogo} from './SidebarLogo';
