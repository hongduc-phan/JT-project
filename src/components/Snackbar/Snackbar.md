Example

```js
const Snackbar = require('./').default;
const {Error} = require('../Icons');

<div>
  <div className="snackbar-row">
    <Snackbar>Company information saved</Snackbar>
  </div>
  <div className="snackbar-row">
    <Snackbar renderIcon={<Error />} variant="error">
      Unable to save company information. Please check your internet connection.
    </Snackbar>
  </div>
</div>;
```
