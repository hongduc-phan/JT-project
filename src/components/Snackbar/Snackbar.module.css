.root {
  padding: 0.875rem 1rem;
  color: var(--white);
  box-shadow: 0 3px 5px -1px rgba(0, 0, 0, 0.2),
    0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 6px 10px 0 rgba(0, 0, 0, 0.14);
  background-color: #202020;
  border-radius: 0.25rem;
  align-items: center;
  display: inline-flex;
  min-width: 349px;
}

.error {
  background-color: #ae2626;
  color: var(--white);
}

.success {
  background-color: var(--green-dark);
  color: var(--white);
}

.icon {
  font-size: 1.5rem;
  margin-right: 0.5rem;
  display: inline-flex;
  align-self: flex-start;
}
