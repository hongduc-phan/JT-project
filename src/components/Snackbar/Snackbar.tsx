import React, {
  AllHTMLAttributes,
  Fragment,
  FunctionComponent,
  ReactNode,
} from 'react';
import clsx from 'clsx';

import Typo, {TypoVariants, TypoColors} from '../Typo';

import styles from './Snackbar.module.css';

export enum SnackbarVariant {
  Error = 'error',
  Success = 'success',
}

export interface SnackbarProps extends AllHTMLAttributes<HTMLDivElement> {
  variant?: SnackbarVariant;
  children: ReactNode;
  color?: TypoColors;
  renderIcon?: ReactNode;
}

const Snackbar: FunctionComponent<SnackbarProps> = ({
  variant,
  children,
  className,
  color,
  renderIcon,
  ...rest
}: SnackbarProps) => {
  return (
    <Typo<HTMLDivElement>
      {...rest}
      tag="div"
      variant={TypoVariants.body2}
      className={clsx(
        styles.root,
        variant === SnackbarVariant.Error && styles.error,
        variant === SnackbarVariant.Success && styles.success,
        className,
      )}
      color={color}
    >
      {renderIcon && (
        <Fragment>
          <span className={styles.icon}>{renderIcon}</span>
          <span>{children}</span>
        </Fragment>
      )}
      {!renderIcon && children}
    </Typo>
  );
};

export default Snackbar;
