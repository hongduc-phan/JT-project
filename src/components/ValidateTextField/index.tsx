export {default} from './ValidateTextField';
export * from './ValidateTextField';
