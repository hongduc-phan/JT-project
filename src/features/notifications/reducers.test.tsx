import {
  notificationActions as actions,
  notificationReducers as reducers,
  NotificationsState,
} from './';
import {SnackbarVariant} from '../../components/Snackbar';

const getInitialState = (initial?: Partial<NotificationsState>) =>
  reducers(initial as NotificationsState, {} as any);

describe('Notifications Reducers', () => {
  describe('Initial state', () => {
    it('Should match snapshot', () => {
      expect(getInitialState()).toMatchSnapshot();
    });
  });

  describe('Snacks', () => {
    it('Can add new snack', () => {
      const initialState = getInitialState();
      expect(initialState.snacks).toHaveLength(0);
      let state = reducers(initialState, actions.snackAdd('message'));
      expect(state.snacks).toHaveLength(1);
      state = reducers(
        state,
        actions.snackAdd('message', SnackbarVariant.Error),
      );
      expect(state.snacks).toHaveLength(2);
      expect(state.snacks[1].variant).toEqual(SnackbarVariant.Error);
    });

    it('Can remove snack', () => {
      const initialState = getInitialState();
      expect(initialState.snacks).toHaveLength(0);
      let state = reducers(initialState, actions.snackAdd('message'));
      state = reducers(
        state,
        actions.snackAdd('message', SnackbarVariant.Error),
      );
      state = reducers(state, actions.snackRemove(state.snacks[0].id));
      expect(state.snacks).toHaveLength(1);
      expect(state.snacks[0].variant).toEqual(SnackbarVariant.Error);
    });
  });
});
