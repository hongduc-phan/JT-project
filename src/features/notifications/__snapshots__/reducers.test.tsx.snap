// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Reducers Initial state Should match snapshot 1`] = `
Object {
  "snacks": Array [],
}
`;
